docs/contributing.md

```diff
@@ -0,0 +1,42 @@
+# Contributing
+
+## Setup
+
+nbgitpuller is a [jupyter
+serverextension](https://jupyter-notebook.readthedocs.io/en/stable/extending/handlers.html),
+and hence can be developed locally without needing a JupyterHub.
+
+1. Setup a virtual environment to do development in
+
+   ```bash
+   python3 -m venv venv
+   source venv/bin/activate
+   ```
+
+2. Install nbgitpuller with its dependencies in this virtual environment
+
+   ```bash
+   pip install -e .
```

Comment:
Those proceeding line-by-line will need to `git clone` this repository first, then cd into the repo directory.

Reply:
@ryanlovett done!